### PR TITLE
feat: use travis stages to run tests with and without a lockfile

### PR DESCRIPTION
**Pull request by lennyburdette**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -36,14 +25,34 @@ before_install:
...
-after_script:
-  - $AFTER_SCRIPT
+  - yarn install --non-interactive
+
+jobs:
+  fail_fast: true
+
+  include:
+    # runs tests with current locked deps and linting
+    - env: NAME=test # used only to make Travis UI show description
```
Comment (rwjblue) [inline .travis.yml]: Lets have this be the main `test` stage, and then make the other things be `additional tests` stage (no need to run them if we don't have non-floating chrome only tests passing)...

File: .travis.yml [hunk 2]
```diff
@@ -36,14 +25,34 @@ before_install:
...
+        - ember sauce:disconnect
+
+    # Run tests again without pinning dependencies to catch issue with drift
+    - env: NAME=floating dependencies
+      install: yarn install --no-lockfile --non-interactive
+      script:
+        - yarn test:node
+        - yarn test
+
+    - env: NAME=floating dependencies sauce
```
Comment (rwjblue) [inline .travis.yml]: we can probably remove this one (I think checking floating deps with chrome only is probably enough)

**Review by rwjblue — CHANGES_REQUESTED** [review]
Looks really nice, made a few inline comments but I'd love to move forward...

**Comment by lennyburdette** [comment]
I had to regenerate the lockfile since our private registry had cached something that's been yanked from npmjs.org. Looks like it works now tho!

File: .travis.yml [hunk 3]
```diff
@@ -36,14 +25,34 @@ before_install:
...
+    # runs tests with current locked deps and linting
+
+    - env: NAME=test # used only to make Travis UI show description
+      script:
+        - yarn test:node
+        - yarn test
 
-before_script:
-  - $BEFORE_SCRIPT
+    - env: NAME=test sauce
```
Comment (rwjblue) [inline .travis.yml]: Lets move this one into additional tests too.

Basically, I want a single job that runs in the first stage which is using the lockfile (e.g. no drift), then do the other two things in "additional tests"...

**Review by rwjblue — COMMENTED** [review]
Sorry for the run around @lennyburdette, I think one last tweak and we are good to go!

**Comment by lennyburdette** [comment]
@rwjblue no worries, I should've read your previous comment more closely. All good now!

**Review by rwjblue — APPROVED** [review]

**Comment by rwjblue** [comment]
Awesome work here, thank you! I really like this pattern (it makes it clear where a particular type of failure comes from, avoids wasting CI time, etc). Hopefully we can leverage the same thing in the addon blueprint to bring the awesomeness to all addons automatically....

**Pull request merged**